### PR TITLE
Next + Supabase quick start: fix error message

### PR DESCRIPTION
**Pull request by josueJURE**
### What?
Fix issue marked as first good issue by Supabase maintainers, see discussion [here](https://github.com/supabase/supabase/issues/28200#issuecomment-2256308802).

### Why?
Improved UX by removing confusion described in [this blogpost](https://3willows.github.io/2024-07-25-display-rate-limit/).

### How?

- More descriptive error message.
- Where available, show the "email rate limit exceeded" error message to the user.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: bef15c58a84190afd4341b8839ff10f0e897df61

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by 3willows** [comment]
Just to say that @josueJURE and I worked on this together (my first pair-programming experience), and we are both new to contributing to the community.  

Any feedback or comments would be much appreciated.  I have subscribed to the issue so will be notified of any developments.

File: examples/with-supabase/app/login/page.tsx [hunk 1]
```diff
@@ -45,7 +45,11 @@ export default function Login({
     });
 
     if (error) {
-      return redirect("/login?message=Could not authenticate user");
+      const { code } = error
+      if (code) {
```
Comment (leerob) [inline examples/with-supabase/app/login/page.tsx:49]: Thank you! Could you please run Prettier in the repo for these changes?

**Comment by leerob** [comment]
Following up here, could you take a look at the Prettier (formatting) issue and address the merge conflicts please?

File: examples/with-supabase/app/login/page.tsx (hunk 1 above)
Comment (3willows) [reply examples/with-supabase/app/login/page.tsx:49]: @leerob apologies I understand @josueJURE has had two weeks of busy interviews and may not have been able to attend to this.  I'll be in touch with him to see if there is anything I can help.  I am not sure whether I can take over and submit the PR on my @3willows account.

**Review by ijjk — CHANGES_REQUESTED** [review]
Gonna close this out as it appears to have been updated more recently https://github.com/vercel/next.js/blob/canary/examples/with-supabase/app/actions.ts

Feel free to re-open against latest changes if still relevant!

**Pull request closed without merging**